src/main/java/eventprocessing/models/ReadingAggregator.java

```diff
@@ -0,0 +1,93 @@
+package eventprocessing.models;
+
+import com.google.common.collect.Iterables;
+import eventprocessing.fileservices.CSVFileService;
+
+import java.time.Clock;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ReadingAggregator {
```

Comment:
This class all looks sensible. `assignResponseToBucket` is a bit complicated, but it looks like it's mostly temporary debugging code. Is that right?